cloud-sdk-tool-build-task/cloud-sdk-package.ts

```diff
@@ -0,0 +1,171 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {getDefaultExecOptions} from 'common/exec-options';
+import * as http from 'http';
+import * as os from 'os';
+import * as path from 'path';
+import * as task from 'vsts-task-lib/task';
+import * as toolLib from 'vsts-task-tool-lib/tool';
+
+type ValidPlatform = 'win32'|'linux'|'darwin';
+type ValidArch = 'x64'|'x86'|'ia32';
+
+function isValidPlatform(platform: string): platform is ValidPlatform {
+  return platform === 'win32' || platform === 'linux' || platform === 'darwin';
+}
+
+function isValidArch(arch: string): arch is ValidArch {
+  return arch === 'x64' || arch === 'x86' || arch === 'ia32';
+}
+
+const cloudSdkId = 'google-cloud-sdk';
+const urlRoot = 'http://dl.google.com/dl/cloudsdk/channels/rapid';
+const downloadUrlRoot = `${urlRoot}/downloads`;
+const versionDoc = `${urlRoot}/components-2.json`;
+
+export class CloudSdkPackage {
+  readonly version: string;
+  private toolPath: string;
+
+  constructor(versionString: string, di: {toolLib: typeof toolLib}) {
+    this.version = di.toolLib.cleanVersion(versionString);
+    if (!(this.version && this.version.length > 0)) {
+      throw new Error(`Given version ${versionString} is not a valid version.`);
+    }
+    this.toolPath = di.toolLib.findLocalTool(cloudSdkId, this.version);
+  }
+
+  static async queryLatestVersion(di: {http: typeof http}): Promise<string> {
+    const message = await new Promise<http.IncomingMessage>(
+        (resolve,
+         reject) => { di.http.get(versionDoc, resolve).on('error', reject); });
+
+    const data = await new Promise<string>((resolve) => {
+      let rawData = '';
+      message.on('data', (chunk: string) => rawData += chunk);
+      message.on('end', () => resolve(rawData));
+    });
+    return JSON.parse(data)['version'] as string;
+  }
+
+  static async createPackage(versionSpec?: string, di = {http, task, toolLib}):
+      Promise<CloudSdkPackage> {
+    let version: string;
+    if (!versionSpec) {
+      version = await CloudSdkPackage.queryLatestVersion(di);
+      di.task.debug(`Latest version ${version} selected.`);
+    } else if (di.toolLib.isExplicitVersion(versionSpec)) {
+      version = versionSpec;
+      di.task.debug(`Version ${version} selected.`);
+    } else {
+      throw new Error('Version, if set, must be an explicit version.');
+    }
+    return new CloudSdkPackage(version, di);
+  }
+
+  isCached(): boolean {
+    return this.toolPath && this.toolPath.length > 0 || false;
+  }
+
+  getToolPath(): string { return this.toolPath; }
+
+  async init(allowReporting: boolean, di = {toolLib, task}): Promise<void> {
+    di.toolLib.prependPath(path.join(this.toolPath, 'google-cloud-sdk', 'bin'));
+    await di.task.tool(di.task.which('gcloud'))
+        .line(`config set disable_usage_reporting ${!allowReporting}`)
+        .exec(getDefaultExecOptions());
+  }
+
+  async aquire(allowReporting: boolean,
```

Comment:
This should be `acquire`.

Reply:
I think it will also be useful if these functions have some comments on what they are supposed to do.

Reply:
Fixed spelling. Added comments.